exposestrategy/ambassador.go

```diff
@@ -0,0 +1,189 @@
+	}
+	if pathMode == PathModeUsePath {
+		suffix := path
+		if len(suffix) == 0 {
+			suffix = "/"
+		}
+		path = UrlJoin("/", svc.Namespace, appName, suffix)
+		hostName = s.domain
+		// fullHostName = UrlJoin(hostName, path)
+	}
+
+	exposePort := svc.Annotations[ExposePortAnnotationKey]
+	if exposePort != "" {
+		port, err := strconv.Atoi(exposePort)
+		if err == nil {
+			found := false
+			for _, p := range svc.Spec.Ports {
+				if port == int(p.Port) {
+					found = true
+					break
+				}
+			}
+			if !found {
+				glog.Warningf("Port '%s' provided in the annotation '%s' is not available in the ports of service '%s'",
+					exposePort, ExposePortAnnotationKey, svc.GetName())
+				exposePort = ""
+			}
+		} else {
+			glog.Warningf("Port '%s' provided in the annotation '%s' is not a valid number",
+				exposePort, ExposePortAnnotationKey)
+			exposePort = ""
+		}
+	}
+	// Pick the fist port available in the service if no expose port was configured
+	if exposePort == "" {
+		port := svc.Spec.Ports[0]
+		exposePort = strconv.Itoa(int(port.Port))
+	}
+
+	servicePort, err := strconv.Atoi(exposePort)
+	if err != nil {
+		return errors.Wrapf(err, "failed to convert the exposed port '%s' to int", exposePort)
+	}
+	glog.Infof("Exposing Port %d of Service %s", servicePort, svc.Name)
+
+	// Here's where we start adding the annotations to our service
+	ambassadorAnnotations := map[string]interface{}{
+		"apiVersion": "ambassador/v1",
+		"kind":       "Mapping",
+		"host":       hostName,
+		"name":       fmt.Sprintf("%s_mapping", hostName),
+		"service":    fmt.Sprintf("%s.%s:%s", appName, svc.Namespace, strconv.Itoa(servicePort))}
+
+	joinedAnnotations := new(bytes.Buffer)
+	fmt.Fprintf(joinedAnnotations, "---\n")
+	yamlAnnotation, err := yaml.Marshal(&ambassadorAnnotations)
+	if err != nil {
+		return err
+	}
+	fmt.Fprintf(joinedAnnotations, "%s", string(yamlAnnotation))
+
+	if s.tlsAcme && svc.Annotations["jenkins-x.io/skip.tls"] != "true" {
+		// we need to prepare the tls module config
+		ambassadorAnnotations = map[string]interface{}{
+			"apiVersion": "ambassador/v1",
+			"kind":       "Module",
+			"name":       "tls",
+			"config": map[string]interface{}{
+				"server": map[string]interface{}{
+					"enabled": "True",
+					"secret":  "tls-" + appName}}}
+
+		yamlAnnotation, err = yaml.Marshal(&ambassadorAnnotations)
+		if err != nil {
+			return err
+		}
+
+		fmt.Fprintf(joinedAnnotations, "---\n")
+		fmt.Fprintf(joinedAnnotations, "%s", string(yamlAnnotation))
+	}
+
+	svc.Annotations["getambassador.io/config"] = joinedAnnotations.String()
+
+	_, err = s.client.Services(svc.Namespace).Update(svc)
+	if err != nil {
+		// yay!
+		return nil
+	} else {
+		return errors.Wrapf(err, "failed to patch the service %s/%s", svc.Namespace, appName)
+	}
+}
+
+func (s *AmbassadorStrategy) Remove(svc *api.Service) error {
+	delete(svc.Annotations, "getambassador.io/config")
+
+	_, err := s.client.Services(svc.Namespace).Update(svc)
+	if err != nil {
```

Comment:
Should this be `==` instead of `!=`?

Reply:
great catch - lemme fix that!

Reply:
PR to fix that here: https://github.com/jenkins-x/exposecontroller/pull/170